device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala

```diff
@@ -8,49 +8,58 @@ import io.circe.generic.auto._
 import org.genivi.sota.data.GroupInfo.Name
 import org.genivi.sota.data.{Namespace, Uuid}
 import org.genivi.sota.device_registry.common.CreateGroupRequest
-import org.genivi.sota.http.UuidDirectives.{extractUuid, allowExtractor}
 import org.genivi.sota.marshalling.RefinedMarshallingSupport._
-import scala.concurrent.{ExecutionContext, Future}
+import org.genivi.sota.http.UuidDirectives.{allowExtractor, extractUuid}
 import slick.driver.MySQLDriver.api._
-
+import org.genivi.sota.device_registry.db._
+import scala.concurrent.{ExecutionContext, Future}
+import org.genivi.sota.marshalling.CirceMarshallingSupport._
+import scala.util.{Failure, Success}
 
 class GroupsResource(namespaceExtractor: Directive1[Namespace])
                     (implicit ec: ExecutionContext,
                     db: Database)
   extends Directives {
 
-  import org.genivi.sota.marshalling.CirceMarshallingSupport._
-  import StatusCodes.Created
-
   private val extractGroupId = allowExtractor(namespaceExtractor, extractUuid, deviceAllowed)
 
   private def deviceAllowed(groupId: Uuid): Future[Namespace] = {
     db.run(GroupInfoRepository.groupInfoNamespace(groupId))
   }
 
   def createGroupFromDevices(request: CreateGroupRequest, namespace: Namespace): Route = {
-    val dbIo = for {
-      info1 <- SystemInfo.findByUuid(request.device1)
-      info2 <- SystemInfo.findByUuid(request.device2)
-    } yield JsonComparison.getCommonJson(info1, info2)
-
-    onSuccess(db.run(dbIo)){
-        case Json.Null =>
-          complete(StatusCodes.BadRequest -> "Devices have no common attributes to form a group")
-        case json =>
-          val groupId = Uuid.generate()
-          val dbIO = for {
-            _       <- GroupInfoRepository.create(groupId, request.groupName, namespace, json)
-            _       <- GroupMember.createGroup(groupId, namespace, request.device1)
-            _       <- GroupMember.createGroup(groupId, namespace, request.device2)
-                 //PRO-1378 Add logic to find devices which should be in this group
-          } yield groupId
-          complete(db.run(dbIO.transactionally))
+
+    val devicesExist = for {
+      _     <- DeviceRepository.exists(namespace, request.device1)
+      _     <- DeviceRepository.exists(namespace, request.device2)
+    } yield ()
```

Comment:
Instead of having the check here, create a extract CreateGroupRequest that checks that the devices belong to the proper namespace.